packages/test/src/mock-native-worker.ts

```diff
@@ -192,6 +193,6 @@ export function isolateFreeWorker(options: WorkerOptions = defaultOptions): Work
         /* Nothing to destroy */
       },
     },
-    compileWorkerOptions(addDefaultWorkerOptions(options))
+    compileWorkerOptions(options, logger)
   );
```

Comment:
I saw that addDefaultWorkerOptions(options,logger) is now called inside compileWorkerOptions. Is compileWorkerOptions internal only? Can we avoid adding defaults or it does not makes sense? Do we need to document this?

Reply:
These are not public APIs.

>  Can we avoid adding defaults or it does not makes sense? 

The `addDefault...Options` vs `compile...Options` is simply a pattern that was adopted internally early on while developing the SDK. There is no reason for one to call either of these without calling the other.

Reply:
I think the original justification for this pattern was to make the code easier to understand and review… `addDefault` would use an object spread operator to set all missing default options, and `compile` do conversion/object initializations where required. That's clean.

Yet, as time passes, I find this pattern sometime makes it harder to introduce changes to options without breaking backward compatibility, and consequently, concerns that ought to be specific to either one are starting to bleed on the other one. Hence, I'm inclined toward hiding this duality from the caller's perspective, and simply make this an internal implementation detail of the `compile...Options` functions, like I did here.